frameworks/kafka/src/main/dist/health-check.sh.mustache

```diff
@@ -0,0 +1,130 @@
+    {{/SECURITY_KERBEROS_ENABLED_FOR_ZOOKEEPER}}
+    
+    export KAFKA_OPTS="-Djava.security.auth.login.config=/tmp/kafka_health_check_client_jaas.conf -Djava.security.krb5.conf=$MESOS_SANDBOX/{{KAFKA_VERSION_PATH}}/config/krb5.conf"
+    {{#SECURITY_KERBEROS_DEBUG}}
+    #export KAFKA_OPTS="-Dsun.security.krb5.debug=true $KAFKA_OPTS"
+    {{/SECURITY_KERBEROS_DEBUG}}
+    
+    {{#SECURITY_TRANSPORT_ENCRYPTION_ENABLED}}
+    SECURITY_PROTOCOL=SASL_SSL
+    {{/SECURITY_TRANSPORT_ENCRYPTION_ENABLED}}
+    {{^SECURITY_TRANSPORT_ENCRYPTION_ENABLED}}
+    SECURITY_PROTOCOL=SASL_PLAINTEXT
+    {{/SECURITY_TRANSPORT_ENCRYPTION_ENABLED}}
+    
+    KAFKA_PRODUCER_CONFIG_OPTIONS="--producer-property sasl.mechanism=GSSAPI --producer-property security.protocol=$SECURITY_PROTOCOL --producer-property sasl.kerberos.service.name=kafka"
+    KAFKA_CONSUMER_CONFIG_OPTIONS="--consumer-property sasl.mechanism=GSSAPI --consumer-property security.protocol=$SECURITY_PROTOCOL --consumer-property sasl.kerberos.service.name=kafka"
+  {{/SECURITY_KERBEROS_ENABLED}}
+  
+  if ! ./{{KAFKA_VERSION_PATH}}/bin/kafka-topics.sh --list --zookeeper $KAFKA_ZOOKEEPER_URI | grep -q $KAFKA_HEALTH_CHECK_TOPIC_PREFIX$POD_INSTANCE_INDEX
+  then
+    ./{{KAFKA_VERSION_PATH}}/bin/kafka-topics.sh --create \
+      --zookeeper $KAFKA_ZOOKEEPER_URI --topic $KAFKA_HEALTH_CHECK_TOPIC_PREFIX$POD_INSTANCE_INDEX \
+      --replica-assignment $POD_INSTANCE_INDEX 2>&1
+  fi
+  
+  # Create a random message
+  random_message=$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 13 ; echo '')
+  echo $random_message > /tmp/kafka_health_check_msg
+  
+  # Publish the message to topic
+  if [ $( ./{{KAFKA_VERSION_PATH}}/bin/kafka-console-producer.sh $KAFKA_PRODUCER_CONFIG_OPTIONS \
+    --broker-list $KAFKA_BROKER_ADDRESS:$KAFKA_BROKER_PORT \
+    --topic $KAFKA_HEALTH_CHECK_TOPIC_PREFIX$POD_INSTANCE_INDEX 2>&1 < /tmp/kafka_health_check_msg | grep -c ">>" ) -ne 1 ]
+  then
+    echo "Health check falied as the message cannot be published! topic: $KAFKA_HEALTH_CHECK_TOPIC_PREFIX$POD_INSTANCE_INDEX msg: $random_message"
+    exit 1
+  else
+    CHECK_CONSUMER=false
+    {{#SECURITY_KERBEROS_ENABLED}}
+    CHECK_CONSUMER=true
+    {{/SECURITY_KERBEROS_ENABLED}}
+    {{#SECURITY_TRANSPORT_ENCRYPTION_ENABLED}}
+    CHECK_CONSUMER=true
+    {{/SECURITY_TRANSPORT_ENCRYPTION_ENABLED}}
+    if $CHECK_CONSUMER
+    then 
+      if ./{{KAFKA_VERSION_PATH}}/bin/kafka-console-consumer.sh $KAFKA_CONSUMER_CONFIG_OPTIONS \
+        --bootstrap-server $KAFKA_BROKER_ADDRESS:$KAFKA_BROKER_PORT \
+        --topic $KAFKA_HEALTH_CHECK_TOPIC_PREFIX$POD_INSTANCE_INDEX \
+        --from-beginning --max-messages 1 2>&1 | grep -q "Processed a total of 1 messages"
```

Comment:
wouldn't it read from start and only the first message?

Reply:
Yes! :smile:

Reply:
but that could be the message that we aren't writing earlier right?

Reply:
Yes that's the case. It's a W+R check when kerberos / ssl is enabled. When they are disabled it is checking the latest message and also the message content.